### ISSUE TITLE
Timewarp --push-exif command failing 

### ISSUE THREAD

**Issue by zephyr325** [RhetTbull/osxphotos#841]
I'm running on Ventura 13.0.1, OSXPhotos 0.54.2, and exiftool 12.50.

I can successfully run this command and changes are applied to the selected picture in the Photos app (I've only selected one picture):

```
osxphotos timewarp --verbose --date 1968-06-01

Updated date/time for photo IMG_20160401_0020.jpg (8484DDC7-D9CB-4703-8AF4-AB745031A64A) from: 1968-08-01 15:09:11 to 1968-06-01 15:09:11
Done.

```
When I add the --push-exif command, to the command, the Traceback I get is:

```
osxphotos timewarp --push-exif --verbose --date 1968-06-01

exiftool path: /usr/local/bin/exiftool

╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/TimCarr/.local/bin/osxphotos:8 in <module>                                                │
│                                                                                                  │
│   5 from osxphotos.__main__ import cli_main                                                      │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(cli_main())                                                                     │
│   9                                                                                              │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/click/core.py:1130 in    │
│ __call__                                                                                         │
│                                                                                                  │
│   1127 │                                                                                         │
│   1128 │   def __call__(self, *args: t.Any, **kwargs: t.Any) -> t.Any:                           │
│   1129 │   │   """Alias for :meth:`main`."""                                                     │
│ ❱ 1130 │   │   return self.main(*args, **kwargs)                                                 │
│   1131                                                                                           │
│   1132                                                                                           │
│   1133 class Command(BaseCommand):                                                               │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/click/core.py:1055 in    │
│ main                                                                                             │
│                                                                                                  │
│   1052 │   │   try:                                                                              │
│   1053 │   │   │   try:                                                                          │
│   1054 │   │   │   │   with self.make_context(prog_name, args, **extra) as ctx:                  │
│ ❱ 1055 │   │   │   │   │   rv = self.invoke(ctx)                                                 │
│   1056 │   │   │   │   │   if not standalone_mode:                                               │
│   1057 │   │   │   │   │   │   return rv                                                         │
│   1058 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!                               │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/click/core.py:1657 in    │
│ invoke                                                                                           │
│                                                                                                  │
│   1654 │   │   │   │   super().invoke(ctx)                                                       │
│   1655 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)                    │
│   1656 │   │   │   │   with sub_ctx:                                                             │
│ ❱ 1657 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub_ctx))               │
│   1658 │   │                                                                                     │
│   1659 │   │   # In chain mode we create the contexts step by step, but after the                │
│   1660 │   │   # base command has been invoked.  Because at that point we do not                 │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/click/core.py:1404 in    │
│ invoke                                                                                           │
│                                                                                                  │
│   1401 │   │   │   echo(style(message, fg="red"), err=True)                                      │
│   1402 │   │                                                                                     │
│   1403 │   │   if self.callback is not None:                                                     │
│ ❱ 1404 │   │   │   return ctx.invoke(self.callback, **ctx.params)                                │
│   1405 │                                                                                         │
│   1406 │   def shell_complete(self, ctx: Context, incomplete: str) -> t.List["CompletionItem"]:  │
│   1407 │   │   """Return a list of completions for the incomplete value. Looks                   │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/click/core.py:760 in     │
│ invoke                                                                                           │
│                                                                                                  │
│    757 │   │                                                                                     │
│    758 │   │   with augment_usage_errors(__self):                                                │
│    759 │   │   │   with ctx:                                                                     │
│ ❱  760 │   │   │   │   return __callback(*args, **kwargs)                                        │
│    761 │                                                                                         │
│    762 │   def forward(                                                                          │
│    763 │   │   __self, __cmd: "Command", *args: t.Any, **kwargs: t.Any  # noqa: B902             │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/osxphotos/cli/timewarp.p │
│ y:652 in timewarp                                                                                │
│                                                                                                  │
│   649 │                                                                                          │
│   650 │   if any([push_exif, pull_exif, function]):                                              │
│   651 │   │   # ExifDateTimeUpdater used to get photo path for --function                        │
│ ❱ 652 │   │   exif_updater = ExifDateTimeUpdater(                                                │
│   653 │   │   │   library_path=library,                                                          │
│   654 │   │   │   verbose=verbose_,                                                              │
│   655 │   │   │   exiftool_path=exiftool_path,                                                   │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/osxphotos/exif_datetime_ │
│ updater.py:63 in __init__                                                                        │
│                                                                                                  │
│    60 │   │   self.db = PhotosDB(self.library_path)                                              │
│    61 │   │   self.verbose = verbose or noop                                                     │
│    62 │   │   self.exiftool_path = exiftool_path                                                 │
│ ❱  63 │   │   self.tzinfo = PhotoTimeZone(library_path=self.library_path)                        │
│    64 │   │   self.plain = plain                                                                 │
│    65 │                                                                                          │
│    66 │   def filename_color(self, filename: str) -> str:                                        │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/osxphotos/phototz.py:48  │
│ in __init__                                                                                      │
│                                                                                                  │
│    45 │   │   if not db_path:                                                                    │
│    46 │   │   │   raise FileNotFoundError("Could not find Photos database path")                 │
│    47 │   │                                                                                      │
│ ❱  48 │   │   photos_version = get_photos_library_version(db_path)                               │
│    49 │   │   db_path = str(pathlib.Path(db_path) / "database/Photos.sqlite")                    │
│    50 │   │   self.db_path = db_path                                                             │
│    51 │   │   self.ASSET_TABLE = _DB_TABLE_NAMES[photos_version]["ASSET"]                        │
│                                                                                                  │
│ /Users/TimCarr/.local/pipx/venvs/osxphotos/lib/python3.11/site-packages/osxphotos/photosdb/photo │
│ sdb_utils.py:143 in get_photos_library_version                                                   │
│                                                                                                  │
│   140 │   │   return 6                                                                           │
│   141 │   if _PHOTOS_7_MODEL_VERSION[0] <= model_ver <= _PHOTOS_7_MODEL_VERSION[1]:              │
│   142 │   │   return 7                                                                           │
│ ❱ 143 │   raise UnknownLibraryVersion(f"db_ver = {db_ver}, model_ver = {model_ver}")             │
│   144                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
```
I'm betting this is something related to the new stuff in Ventura/Photos database.

**Comment by zephyr325** [comment]
Great - thank you!

**Comment by RhetTbull** [comment]
@all-contributors please add @zephyr325 for bug

**Comment by RhetTbull** [comment]
I can't test it on Ventura but I believe this should be fixed in [v0.54.3](https://github.com/RhetTbull/osxphotos/releases/tag/v0.54.3)

**Comment by zephyr325** [comment]
Confirmed to now be working correctly in 0.54.3.  Let me know if you need any other Ventura testing.
